text/0000-subslice-pattern-syntax.md

```diff
@@ -0,0 +1,162 @@
+a subslice and can parse following tokens as a pattern (unless they are `,` or `]`, then it's just
+`..`, without an attached pattern).  
+With `PAT..` we need to consume the pattern first, but that pattern may be a... `RANGE_BEGIN..`
+range pattern, then it means that we consumed too much and need to reinterpret the parsed tokens
+somehow. It's probably possible to make this work, but it's some headache that we would like to
+avoid if possible.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Subslice (aka "rest of the slice") in a slice patterns can be matched to a pattern `PAT` using
+syntax `..PAT`.  
+`..` with the pattern omitted is a sugar for `.._` (wildcard pattern) so it means
+"ignore the rest of the slice".
+
+Example (without `feature(match_default_bindings)`):
+```rust
+let v = vec![1, 2, 3];
+match v[..] {
+    [1, ..ref subslice, 4] => assert_eq!(subslice.len(), 1),
+    [5, ..ref subslice] => assert_eq!(subslice.len(), 2),
+    [..ref subslice, 6] => assert_eq!(subslice.len(), 2),
+    [x, .., y] => assert!(v.len() >= 2),
+    [..] => {} // Always matches
+}
+```
+Example (with `feature(match_default_bindings)`):
+```rust
+let v = vec![1, 2, 3];
+match &v[..] {
+    [1, ..subslice, 4] => assert_eq!(subslice.len(), 1),
+    [5, ..subslice] => assert_eq!(subslice.len(), 2),
+    [..subslice, 6] => assert_eq!(subslice.len(), 2),
+    [x, .., y] => assert!(v.len() >= 2),
+    [..] => {} // Always matches
+}
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Subslice in a slice patterns can be matched to a pattern `PAT` using syntax `..PAT`.  
+`..` with the pattern omitted is a sugar for `.._`.
+
+If ambiguity with some other syntactic construction arises in the future, disambiguation will be
+performed in favor of the subslice pattern.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+None known.
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+The `PAT..` alternative was discussed in the motivational part of the RFC.
+
+More complex syntaxes derived from `..` are possible, they use additional tokens to avoid the
+ambiguity with ranges, for example
+[`..PAT..`](https://github.com/rust-lang/rust/issues/23121#issuecomment-301485132), or
+`.. @ PAT` or `PAT @ ..` (original comments seem to be lost by GitHub), or other similar
+alternatives.  
+We reject these syntaxes because they only bring benefits in incredibly contrived cases using a
+feature that doesn't even exist yet, but normally they only add symbolic noise.
+
+More radical syntax changes not keeping consistency with `..`, for example
+[`[1, 2, 3, 4] ++ ref v`](https://github.com/rust-lang/rust/issues/23121#issuecomment-289220169).
+
+# Prior art
+[prior-art]: #prior-art
+
+Some other languages like Scala or F# has list/array patterns, but their
+syntactic choices are quite different from Rust's general style.
+
```

Comment:
Another language you might want to mention / look at is Haskell:
+ https://stackoverflow.com/questions/35378922/pattern-match-list-with-exactly-2-elements-in-haskell
+ https://en.wikibooks.org/wiki/Haskell/Pattern_matching